pyiceberg/catalog/dynamodb.py

```diff
@@ -152,6 +156,14 @@ def create_table(
             ValueError: If the identifier is invalid, or no path is given to store metadata.
 
         """
+        if not isinstance(schema, Schema):
+            import pyarrow as pa
+
+            from pyiceberg.io.pyarrow import _ConvertToIcebergWithFreshIds, pre_order_visit_pyarrow
+
+            if isinstance(schema, pa.Schema):
+                schema: Schema = pre_order_visit_pyarrow(schema, _ConvertToIcebergWithFreshIds())  # type: ignore
```

Comment:
Should we also catch the exceptions when we cannot convert the schema? Also, I think we need to add a try-except when PyArrow is not available, and we can also raise an appropriate exception.
```suggestion
            try:
                import pyarrow as pa

                from pyiceberg.io.pyarrow import _ConvertToIcebergWithFreshIds, pre_order_visit_pyarrow

                if isinstance(schema, pa.Schema):
                    schema: Schema = pre_order_visit_pyarrow(schema, _ConvertToIcebergWithFreshIds())  # type: ignore
            except ModuleNotFoundError:
                # throw an appropriate error here
```

Reply:
Maybe we should also move this to a helper function in `pyarrow.py`?

Reply:
I thought about that as well, but I think the main verbiage of this code block is in avoiding pyarrow imports. If we put this block of code in pyarrow.py, I think we end up importing pyarrow, or we would end up with another:

```
if not isinstance(schema, Schema):
          try:
                import pyarrow as pa
                from pyiceberg.io.pyarrow import _new_schema_from_pyarrow_schema
                if isintance(schema, pa.Schema):
                     schema: Schema = _new_schema_from_pyarrow_schema(schema)
                     
...
```

So if we wanted this code block to be defined as a function, we would have to put it in a separate module where the base module imports don't include pyarrow

Reply:
Alright, maybe then a function in `catalog/__init__.py` that will just do the check:

```python
def _convert_schema_if_needed(schema: Union[Schema, "pa.Schema"]) -> Schema:
    try:
        import pa
        # Do the conversion with all the imports 
    except ModuleNotFoundError:
        pass

    if not isinstance(schema, Schema):
        raise ValueError(..)
    return schema

```

Reply:
Makes sense @Fokko 🥇